src/util/locking/RedisResourceLocker.ts

```diff
@@ -0,0 +1,113 @@
+import type { RedisClient } from 'redis';
+import { createClient } from 'redis';
+import type { Lock } from 'redlock';
+import Redlock from 'redlock';
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import { getLoggerFor } from '../../logging/LogUtil';
+import { InternalServerError } from '../errors/InternalServerError';
+import type { ResourceLocker } from './ResourceLocker';
+
+/**
+ * A locking system that uses a Redis server or any number of
+ * Redis nodes / clusters
+ */
+export class RedisResourceLocker implements ResourceLocker {
+  protected readonly logger = getLoggerFor(this);
+
+  protected readonly clients: RedisClient[];
+  public readonly redlock: Redlock;
+
+  private readonly lockList: Map<string, Lock>;
+
+  public constructor(redisclients: string[]) {
+    this.lockList = new Map();
+    this.clients = [];
+    if (redisclients) {
+      redisclients.forEach((client: string): void => {
+        // Check if portnumber or ip with portnumber
+        // Defenitely not perfect, but configuring this is only for experienced users
+        if (!/(.+:)?[0-9]{4,5}/u.test(client)) {
+          throw new Error(`Invalid data provided to create a Redis client: ${client}`);
+        }
+        const split = client.split(':');
+        let res: RedisClient;
+        if (split && split.length === 1) {
+          // Only a portnumber was provided
+          res = createClient(Number(split[0]));
+        } else if (split && split.length === 2) {
+          // Portnumber and host were provided
+          res = createClient(Number(split[1]), split[0]);
+        } else {
+          throw new Error(`Invalid data provided to create a Redis client: ${client}`);
+        }
+        this.clients.push(res);
+      });
+    }
+    this.redlock = new Redlock(
+      this.clients,
+      {
+        // The expected clock drift; for more details
+        // see http://redis.io/topics/distlock
+        // Multiplied by lock ttl to determine drift time
+        driftFactor: 0.01,
+        // The max number of times Redlock will attempt
+        // to lock a resource before erroring
+        retryCount: 10,
+        // The time in ms between attempts
+        retryDelay: 200,
+        // The max time in ms randomly added to retries
+        // to improve performance under high contention
+        // see https://www.awsarchitectureblog.com/2015/03/backoff.html
+        // time in ms
+        retryJitter: 200,
+      },
+    );
+    this.redlock.on('clientError', (err): void => {
+      this.logger.error('A Redis/Redlock error occured', err);
+    });
+  }
+
+  public async acquire(identifier: ResourceIdentifier): Promise<void> {
+    const resource = identifier.path;
+    const ttl = 1000;
```

Comment:
This means the lock will expire after 1 second right? We don't want this (and if it's there we definitely want to throw an error when this expiration happens) since we have another component taking part of the expiration. If we have to provide an amount it should be a lot higher than 1s. The lock should also be removed from the locklist on timeout.

@RubenVerborgh This could be a case for changing the locker interface a bit (not in this PR though) since we have an `ExpiringReadWriteLocker`, but not an `ExpiringResourceLocker` (which is what this class is).

Reply:
We could change this to a ridiculously high number?